lib/passport/callbacks.coffee

```diff
@@ -130,6 +130,7 @@ onAccessToken = (req, done, params) -> (err, res) ->
   # /user API. Then attempt to fetch the access token again from Gravity and
   # recur back into this onAcccessToken callback.
   else if msg.match('no account linked')?
+    if (req?.session?.signupIntent && params?) then params.signup_intent = req.session.signupIntent
```

Comment:
maybe better as `params.signup_intent = req.session.signupIntent if req?.session?.signupIntent and params?`

Reply:
Oh, I didn't know you oculd do that sort of backwards conditional. was going to change it to `if (...) then ...` - either way!